python/ray/train/base_trainer.py

```diff
@@ -726,8 +671,12 @@ def train_func(config):
             # else: Train will restore from the user-provided
             # `resume_from_checkpoint` == `starting_checkpoint`.
 
+            # Evaluate datasets if they are wrapped in a factory.
+            trainer.datasets = {
+                k: d() if callable(d) else d for k, d in self.datasets.items()
+            }
+
             trainer.setup()
-            trainer.preprocess_datasets()
             trainer.training_loop()
```

Comment:
`preprocess_datasets` used to do this "evaluation" of the dataset factory function, plus any dataset preprocessing from `Trainer.preprocessor`. This interface is not needed so I removed it.